### PR TITLE
fix(Button): Update text button styles

### PR DESCRIPTION
**Pull request by codepretty**
Updated text button styles to match redlines and have the correct hover/focus states. Also, reorganized some of the buttons doc page.

**BEFORE**
image

**AFTER**
image

**Dark theme**
image

**Contrast theme**
image

### CODE REVIEW

File: docs/src/examples/components/Button/Types/ButtonExampleText.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react'
+import { Button, Text } from '@stardust-ui/react'
+
+const ButtonExampleText = () => (
+  <Button text>
+    <Text content="A text button" />
```
Comment (mnajdova) [inline docs/src/examples/components/Button/Types/ButtonExampleText.tsx]: Can we have one example with primary variation of the text button?

File: packages/react/src/themes/teams/components/Button/buttonStyles.ts [hunk 2]
```diff
@@ -139,21 +139,30 @@ const buttonStyles: ComponentSlotStylesInput<ButtonProps & ButtonState, ButtonVa
         color: textColor,
         backgroundColor: 'transparent',
         borderColor: 'transparent',
+        padding: `0 ${pxToRem(8)}`,
+
+        // by default icons should always be outline on rest and filled on hover/focus
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Button/buttonStyles.ts]:
```suggestion
        // by default icons should always be outline, but filled on hover/focus
```

File: packages/react/src/themes/teams-dark/components/Button/buttonVariables.ts [hunk 3]
```diff
@@ -24,10 +24,5 @@ export default (siteVars: any): Partial<ButtonVariables> => {
     circularBackgroundColorHover: siteVars.colors.grey[300],
     circularBackgroundColorFocus: siteVars.colors.grey[250],
     circularBorderColor: siteVars.colors.grey[250],
-
```
Comment (mnajdova) [inline packages/react/src/themes/teams-dark/components/Button/buttonVariables.ts:27]: 👍

File: docs/src/examples/components/Button/Types/ButtonExampleIconOnly.shorthand.tsx [hunk 4]
```diff
@@ -1,6 +1,17 @@
...
-const ButtonExampleIconOnly = () => <Button icon="book" iconOnly primary />
+const ButtonExampleIconOnly = () => (
+  <div>
+    <Flex gap="gap.large" vAlign="center">
+      <Button icon="stardust-close" iconOnly />
+      <Text content="AS A DEFAULT BUTTON" weight="bold" />
+    </Flex>
+    <Flex gap="gap.large" vAlign="center">
+      <Button icon="stardust-close" text iconOnly />
+      <Text content="AS A TEXT BUTTON" weight="bold" />
```
Comment (mnajdova) [inline docs/src/examples/components/Button/Types/ButtonExampleIconOnly.shorthand.tsx:12]: Wouldn't it be better to have all text variations in one example (default, primary, iconOnly etc...) What was the reason these were separated across different examples?

Comment (codepretty) [reply docs/src/examples/components/Button/Types/ButtonExampleIconOnly.shorthand.tsx:12]: Yes, good callout. I added them.

**Review by alinais — APPROVED** [review]

**Pull request merged**